pkg/persistence/postgrespersister.go

```diff
@@ -185,6 +185,15 @@ func (p *PostgresPersister) retrieveEventsQuery(tableName string, criteria *mode
 			p.addWhereAnd(queryBuf)
 			queryBuf.WriteString(" event_type = :eventtype") // nolint: gosec
 		}
+		if criteria.ContractAddress != "" {
+			p.addWhereAnd(queryBuf)
+			queryBuf.WriteString(" contract_address = :contract_address") // nolint: gosec
+		}
+		if len(criteria.ExcludeHashes) > 0 {
+			p.addWhereAnd(queryBuf)
+			notInQuery := fmt.Sprintf(" hash NOT IN ('%v')", strings.Join(criteria.ExcludeHashes, "','"))
```

Comment:
I would also run an EXPLAIN query on this just to make sure the indices are being used, not sure how NOT IN clause works in postgres.

Reply:
Nice Pete, you're absolutely right, not in does a seq scan will change.